openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMUtils.java

```diff
@@ -57,29 +60,51 @@ static public void loadLibs() {
 
         if (!libsLoaded) {
             try {
-                loadSharedLibraryFromJar("libgomp.so.1.0.0");
-                loadSharedLibraryFromJar("lib_lightgbm.so");
-                loadSharedLibraryFromJar("lib_lightgbm_swig.so");
-            } catch (final IOException e) {
-                throw new RuntimeException("Failed to load LightGBM shared libraries from jar.", e);
+                final CpuArchitecture cpuArchitecture = getCpuArchitecture(System.getProperty("os.arch"));
+                loadSharedLibraryFromJar("libgomp.so.1.0.0", cpuArchitecture);
+                loadSharedLibraryFromJar("lib_lightgbm.so", cpuArchitecture);
+                loadSharedLibraryFromJar("lib_lightgbm_swig.so", cpuArchitecture);
+            } catch (final Exception ex) {
+                throw new RuntimeException("Failed to load LightGBM shared libraries from jar.", ex);
             }
 
             logger.info("Loaded LightGBM libs.");
             libsLoaded = true;
         }
     }
 
+    static CpuArchitecture getCpuArchitecture(final String cpuArchName) {
+        final String upperCaseCpuArchName = cpuArchName.toUpperCase();
+        final boolean knownCpuArchitecture = Arrays.stream(CpuArchitecture.values())
```

Comment:
We can simply catch the `IllegalArgumentException` thrown by the  `CpuArchitecture.valueOf(upperCaseCpuArchName)` and work from there no? It is simpler I guess.

Reply:
it is. done